docs/configuration.md

```diff
@@ -29,22 +29,54 @@ reason = "No external http servers are written in Go lang."
 
 Ignoring a vulnerability will also ignore vulnerabilities that are considered aliases of that vulnerability.
 
-## Override specific package
+## Override packages
 
-To ignore a specific a package, or manually set its license, enter the package name and ecosystem under the `PackageOverrides` key.
+You can specify overrides for particular packages to have them either ignored entirely or to set their license using the `PackageOverrides` key:
 
 ```toml
 [[PackageOverrides]]
-# The package name, version, and ecosystem to match against
+# One or more fields to match each package against:
 name = "lib"
-# If version is not set or empty, it will match every version
 version = "1.0.0"
 ecosystem = "Go"
-# Ignore this package entirely, including license scanning
+group = "dev"
+
+# Actions to take for matching packages:
+ignore = true # Ignore this package entirely, including license scanning
+license.override = ["MIT", "0BSD"] # Override the license of the package, if it is not ignored
+
+effectiveUntil = 2022-11-09 # Optional exception expiry date, after which the override will no longer apply
+reason = "abc" # Optional reason for the override, to explain why it was added
+```
+
+Each override entry will be checked against every package being scanned, and applied if all the configured fields match. This enables both very broad and very specific overrides based on your needs:
```

Comment:
When ignoring vulnerabilities, we only apply the config to the lockfiles that is in the same directory, is this the same behavior here? Or is it applying to everything?

Reply:
I've not touched the logic that handles deciding if it should check if an override applies for a package (none of the code I touched dealt with paths, among other things) - so that behaviour should remain unchanged, and also wasn't mentioned in the original docs.

In saying that, I assume what probably triggered this was the "will be checked against every package being scanned" which I agree could make people think it applies to... well _every_ package being scanned 😅 

I'll reword this to remove that ambiguity

Reply:
Just had another look at the logic because I wasn't too sure what the current behavior is, but yes you're right we do only affect lockfiles in the same directory as the ignore file.

(we probably can add some tests specifically for nested lockfiles+configs, though it'll conflict with our osv-scanner-action and scorecard..., I'll add an issue for this).